objectivec/Tests/GPBWellKnownTypesTest.m

```diff
@@ -99,4 +101,44 @@ - (void)testDuration {
   [duration2 release];
 }
 
+- (void)testAnyToMessageTransitionUtils {
+  TestAllTypes *from = [TestAllTypes message];
+  [self setAllFields:from repeatedCount:1];
+  NSData *data = from.data;
+
+  // Test initWithMessage
+  GPBAny * anyInited = [[GPBAny alloc] initWithMessage:from];
+  XCTAssertEqualObjects(
+      [GPBTypeGoogleApisComPrefix stringByAppendingString:from.descriptor.name],
+      anyInited.typeURL);
+  XCTAssertEqualObjects(data, anyInited.value);
+  [anyInited release];
+
+  // Test setMessage.
+  GPBAny *any = [GPBAny message];
+  [any setMessage:from];
+  XCTAssertEqualObjects(
+      [GPBTypeGoogleApisComPrefix stringByAppendingString:from.descriptor.name],
+      any.typeURL);
+  XCTAssertEqualObjects(data, any.value);
+
+  // Test messageOfClass
+  TestAllTypes *to = (TestAllTypes*)[any messageOfClass:[TestAllTypes class]];
+  XCTAssertEqualObjects(from, to);
+  XCTAssertEqual([any messageOfClass:[ForeignMessage class]], nil);
+
+  // Test setMessage with another type.
+  ForeignMessage *from2 = [ForeignMessage message];
+  [any setMessage:from2];
+  XCTAssertEqualObjects(
+      [GPBTypeGoogleApisComPrefix stringByAppendingString:from2.descriptor.name],
+      any.typeURL);
+  XCTAssertEqual(0UL, [any.value length]);
+
+  // Test wrapsMessageOfClass
+  XCTAssertTrue([any wrapsMessageOfClass:[from2 class]]);
+  XCTAssertFalse([any wrapsMessageOfClass:[from class]]);
+  XCTAssertThrows([any wrapsMessageOfClass:[NSString class]]);
```

Comment:
See comment above about conditionalizing the unit test based on

#if !defined(NS_BLOCK_ASSERTIONS)